pghoard/restore.py

```diff
@@ -138,19 +163,36 @@ def get_basebackup(self, arg):
         """Download a basebackup from an object store"""
         try:
             self.storage = self._get_object_storage(arg.config, arg.site, arg.target_dir)
-            self._get_basebackup(arg.target_dir, arg.basebackup, arg.site, arg.primary_conninfo, overwrite=arg.overwrite)
+            self._get_basebackup(arg.target_dir, arg.basebackup, arg.site, arg.primary_conninfo,
+                                 recovery_target_time=arg.recovery_target_time,
+                                 recovery_target_xid=arg.recovery_target_xid,
+                                 overwrite=arg.overwrite)
         except Exception as ex:
             raise RestoreError("{}: {}".format(ex.__class__.__name__, ex))
 
-    def _get_basebackup(self, pgdata, basebackup, site, primary_conninfo, overwrite=False):
+    def _find_nearest_basebackup(self, recovery_target_time=None):
+        basebackups = self.storage.list_basebackups()
+        if recovery_target_time:
+            applicable_basebackups = [basebackup["name"] for basebackup in basebackups
+                                      if basebackup["last_modified"] <= recovery_target_time]
+        else:
+            applicable_basebackups = [basebackup["name"] for basebackup in basebackups]
+        if not applicable_basebackups:
+            print("No applicable basebackups found, exiting")
+            sys.exit(-1)
```

Comment:
`sys.exit()` should only be called from the main function, return None instead here and return a statuscode (`1`?) in `_get_basebackup` below.  Alternatively / preferrably raise a custom exception type which gets mapped to a proper error code and error message in the main function.  That'd make it easier to test this.